test/vinyl-luatest/update_optimize_test.lua

```diff
@@ -0,0 +1,405 @@
+    -- 3 = REPLACE in index1 and DELETE + REPLACE in index3.
+    t.assert_equals(new_stmt_count - old_stmt_count, 3)
+    old_stmt_count = new_stmt_count
+    t.assert_equals(
+        index:select{}, {{1, 2, 3, 4, 5},
+                         {30, 3, 9, 30, 4, 100},
+                         {3, 4, 5, 6, 7},
+                         {4, 5, 6, 7, 8},
+                         {10, 20, 30, 40, 500}})
+    t.assert_equals(
+        index2:select{}, {{1, 2, 3, 4, 5},
+                          {3, 4, 5, 6, 7},
+                          {4, 5, 6, 7, 8},
+                          {30, 3, 9, 30, 4, 100},
+                          {10, 20, 30, 40, 500}})
+    t.assert_equals(
+        index3:select{}, {{30, 3, 9, 30, 4, 100},
+                          {1, 2, 3, 4, 5},
+                          {3, 4, 5, 6, 7},
+                          {4, 5, 6, 7, 8},
+                          {10, 20, 30, 40, 500}})
+end
+
+g.test_during_dump_optimizes_update_not_skip_entire_key = function()
+    -- Check if optimizes update do not skip the entire key during dump.
+    t.assert_equals(space:replace{10, 100, 1000, 10000, 100000, 1000000},
+        {10, 100, 1000, 10000, 100000, 1000000})
+    t.assert_equals(
+        index:update({100}, {{'=', 6, 1}}), {10, 100, 1000, 10000, 100000, 1})
+
+    box.begin()
+    t.assert_equals(space:replace{20, 200, 2000, 20000, 200000, 2000000},
+        {20, 200, 2000, 20000, 200000, 2000000})
+    t.assert_equals(
+        index:update({200}, {{'=', 6, 2}}), {20, 200, 2000, 20000, 200000, 2})
+    box.commit()
+    t.assert_equals(box.snapshot(), 'ok')
+    dump_count = wait_for_dump(index, dump_count)
+    old_stmt_count = dumped_stmt_count(index, index2, index3)
+    t.assert_equals(
+        index:select{}, {{1, 2, 3, 4, 5},
+                         {30, 3, 9, 30, 4, 100},
+                         {3, 4, 5, 6, 7},
+                         {4, 5, 6, 7, 8},
+                         {10, 20, 30, 40, 500},
+                         {10, 100, 1000, 10000, 100000, 1},
+                         {20, 200, 2000, 20000, 200000, 2}})
+    t.assert_equals(
+        index2:select{}, {{1, 2, 3, 4, 5},
+                          {3, 4, 5, 6, 7},
+                          {4, 5, 6, 7, 8},
+                          {30, 3, 9, 30, 4, 100},
+                          {10, 20, 30, 40, 500},
+                          {10, 100, 1000, 10000, 100000, 1},
+                          {20, 200, 2000, 20000, 200000, 2}})
+    t.assert_equals(
+        index3:select{}, {{30, 3, 9, 30, 4, 100},
+                          {1, 2, 3, 4, 5},
+                          {3, 4, 5, 6, 7},
+                          {4, 5, 6, 7, 8},
+                          {10, 20, 30, 40, 500},
+                          {10, 100, 1000, 10000, 100000, 1},
+                          {20, 200, 2000, 20000, 200000, 2}})
+    -- gh-2980: key uniqueness is not checked if indexed fields
+    -- are not updated.
+    space:truncate()
```

Comment:
This should be a different test case. Simply re-create the space instead of truncating it.

Reply:
Done